website/src/documentation/patterns/side-inputs.md

```diff
@@ -50,25 +50,25 @@ For instance, the following code sample uses a `Map` to create a `DoFn`. The `Ma
 
 ## Slowly updating side input using windowing
 
-You can read side input pcollection periodically into distinct windows.
-Later, when you apply side input to your main input, windows will be matched automatically 1:1.
-This way, you can guarantee side input consistency on the duration of the single window.
-
-To do this, you can utilize PeriodicSequence PTransform that will generate infinite sequence
-of elements with some real-time period:
+You can read side input data periodically into distinct PCollection windows.
+Later, when you apply the side input to your main input, each main input
+window is automatically matched to a single side input window.
+This guarantees side input consistency on the duration of the single window,
+meaning that each window on main input side will be matched to a single
+version of side input data.
+
+To do this, you can utilize a combination of PeriodicSequence/PeriodicImpulse
+PTransforms that will generate infinite sequence of elements with some real-time
+period and SDF Read operation or similar to read data into side input
+periodically:
 
 1. Use the PeriodicImpulse transform to generate windowed periodic sequence.
 
-    a. MAX_TIMESTAMP can be replaced with some closer boundary if you want to stop generating elements at some point.
-
 1. Read data using Read operation triggered by arrival of PCollection element.
```

Comment:
On a second look, I think it's redundant to say "Read data using a Read operation" - maybe we can say something like "Read data when a PCollection element arrives"